backend/src/app/domain/components/services.py

```diff
@@ -169,160 +169,112 @@ async def delete_many(
-                pprint(data.elements)
-
-                # data = await super().to_model(data)
-
-                # pprint(data)
-
-                # data = await super().to_model(data)
-                # data = m.Component(**data)
-
-                # data = await super().get(item_id=data["id"])
-
-                # if name is not None:
-                #     data.name = name
-                #
-                # if comment is not None:
-                #     data.comment = comment
-                # if scopes is not None:
-                #     data.scopes = scopes
-
-                # if elements is not None:
-                #     if len(elements) > 0:
-                #         # Create the elements
-                #
-                #         for element in elements:
-                #             element_to_add = m.Element(
-                #                 amount=element.amount,
-                #                 material_id=element.material,
-                #                 component=data,
-                #             )
-
-                # if element.transforms:
-                #     # if (obj := (await session.execute(statement)).scalar_one_or_none()) is None:
-                #     process_transforms = await processes_service.list(
-                #         m.Process.id.in_(element.transforms)
-                #     )
-                #
-                #     if len(element.transforms) != len(process_transforms):
-                #         raise ForeignKeyError(
-                #             detail=f"A foreign key for transforms is invalid {element.transforms} {process_transforms}"
-                #         )
-                #
-                #     element_to_add.process_transforms.extend(
-                #         process_transforms
-                #     )
-
-                # data.elements.append(element_to_add)
-                # self.repository.session.add(element_to_add)
-
-                if owner:
-                    owner.journal_entries.append(
-                        m.JournalEntry(
-                            table_name=m.Component.__tablename__,
-                            record_id=data.id,
-                            action=m.JournalAction.UPDATED,
-                            user=owner,
-                            value=input_data
-                            if is_dict(input_data)
-                            else input_data.to_dict(),
-                        )
-                    )
+        if (
+            operation == "create"
+            and is_dict(data)
+            or (operation == "upsert" and not has_id)
+        ):
+            with self.repository.session.no_autoflush:
+                return await self._create_component(data, processes_service, owner)
+
+        if (
+            operation == "update"
+            and is_dict(data)
+            or (operation == "upsert" and has_id)
+        ):
+            with self.repository.session.no_autoflush:
```

Comment:
I don’t remember for sure right now, but I think it’s the no_autoflush is necessary for the create, but useless for the update.